Origin: https://sources.debian.org/src/chromium/107.0.5304.110-2/debian/patches/fixes/angle-wayland.patch/
description: this is just wayland-client-core.h diffed against /usr/include/wayland-client-core.h in sid
author: Andres Salomon <dilinger@debian.org>

ANGLE upstream broke wayland inclusion stuff when building against system
libwayland. It's fine in bullseye (where libwayland is older), but breaks
in sid. This brings the header in chromium/angle with what's in sid.

Drop this once ANGLE gets updated.


--- a/third_party/wayland/src/src/wayland-client-core.h
+++ b/third_party/wayland/src/src/wayland-client-core.h
@@ -119,9 +119,27 @@ struct wl_display;
  */
 struct wl_event_queue;
 
+/** Destroy proxy after marshalling
+ * @ingroup wl_proxy
+ */
+#define WL_MARSHAL_FLAG_DESTROY (1 << 0)
+
 void
 wl_event_queue_destroy(struct wl_event_queue *queue);
 
+struct wl_proxy *
+wl_proxy_marshal_flags(struct wl_proxy *proxy, uint32_t opcode,
+		       const struct wl_interface *interface,
+		       uint32_t version,
+		       uint32_t flags, ...);
+
+struct wl_proxy *
+wl_proxy_marshal_array_flags(struct wl_proxy *proxy, uint32_t opcode,
+			     const struct wl_interface *interface,
+			     uint32_t version,
+			     uint32_t flags,
+			     union wl_argument *args);
+
 void
 wl_proxy_marshal(struct wl_proxy *p, uint32_t opcode, ...);
 
